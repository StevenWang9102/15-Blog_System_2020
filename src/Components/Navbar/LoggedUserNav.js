import React from "react";
import { Link } from "react-router-dom";
import {displayLimit, offset } from "../../ReduxStore/HttpClient"
import PropTypes from "prop-types";

export const LoggedUserNav = props => {


  const userInfoLocal = props.getUserInformationLocal();

  return (
    <div>
      <li className='nav-item'>
        <Link className='nav-link active' to='/home'>
          Home
        </Link>
      </li>

      <li className='nav-item'>
        <Link
          className='nav-link'
          to='/new_post'
          onClick={() => {
            props.postedArticleReloaded(false);
          }}>
          <img src='../../icon/008-edit.png' alt='' />
          New Post
        </Link>
      </li>

      <li className='nav-item'>
        <Link className='nav-link' to='/setting'>
          <img src='../../icon/004-settings.png' alt='' />
          Setting
        </Link>
      </li>

      <li className='nav-item'>
        <Link
          className='nav-link'
          to={`/user_profile/${userInfoLocal.username}`}
          onClick={() => {
            props.loadUserProfileDetail(userInfoLocal.username, displayLimit, offset);
            props.setProfileNavStatus("active", "null");
          }}>
          <img className='user-pic' src={userInfoLocal.image} alt='' />
          {userInfoLocal.username}
        </Link>
      </li>
    </div>
  );
};

LoggedUserNav.propTypes = {
  getUserInformationLocal: PropTypes.func,
  postedArticleReloaded: PropTypes.func,
  loadUserProfileDetail: PropTypes.func,
  setProfileNavStatus: PropTypes.func,
};