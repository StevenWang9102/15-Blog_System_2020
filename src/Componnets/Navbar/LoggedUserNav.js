import React from "react";
import { Link } from "react-router-dom";
import { getUserFromSession } from "../../Functions/AuthToken";

export const LoggedUserNav = props => {

  const getUserInformationLocal = () => {
    if (
      !props.userInformation ||
      (props.userInformation && !props.userInformation.token)
    ) {
      const userInformationOnSession = getUserFromSession();
      if (userInformationOnSession) {
        props.userInformationLoaded({ user: userInformationOnSession });
      }
      return userInformationOnSession;
    } else {
      return props.userInformation;
    }
  };

  const userInfoLocal = getUserInformationLocal();

  return (
    <div>
      <li className='nav-item'>
        <Link className='nav-link active' to='/home'>
          Home
        </Link>
      </li>

      <li className='nav-item'>
        <Link
          className='nav-link'
          to='/new_post'
          onClick={() => {
            props.postedArticleReloaded(false);
          }}>
          <img src='../icon/008-edit.png' alt='' />
          New Post
        </Link>
      </li>

      <li className='nav-item'>
        <Link className='nav-link' to='/setting'>
          <img src='../icon/004-settings.png' alt='' />
          Setting
        </Link>
      </li>

      <li className='nav-item'>
        <Link
          className='nav-link'
          to={`/user_profile/${userInfoLocal.username}`}
          onClick={() => {
            props.loadUserProfileDetail(userInfoLocal.username, 5, 0);
            props.setProfileNavStatus("active", "null");
            props.setLoading("LOADING");
          }}>
          <img className='user-pic' src={userInfoLocal.image} alt='' />
          {userInfoLocal.username}
        </Link>
      </li>
    </div>
  );
};
