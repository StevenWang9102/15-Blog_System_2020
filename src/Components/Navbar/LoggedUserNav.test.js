import React from "react";
import { LoggedUserNav } from "./LoggedUserNav";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <LoggedUserNav
        getUserInformationLocal={() => {}}
        postedArticleReloaded={() => {}}
        loadUserProfileDetail={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
