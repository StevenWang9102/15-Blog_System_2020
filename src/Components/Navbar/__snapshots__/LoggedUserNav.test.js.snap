// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <li
    className="nav-item"
  >
    <a
      className="nav-link active"
      href="#/home"
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#/new_post"
      onClick={[Function]}
    >
      <img
        alt=""
        src="../../icon/008-edit.png"
      />
      New Post
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#/setting"
      onClick={[Function]}
    >
      <img
        alt=""
        src="../../icon/004-settings.png"
      />
      Setting
    </a>
  </li>
  <li
    className="nav-item"
  >
    <a
      className="nav-link"
      href="#/user_profile/undefined"
      onClick={[Function]}
    >
      <img
        alt=""
        className="user-pic"
      />
    </a>
  </li>
</div>
`;
